### PR TITLE
fix(Activity): Fix equals() not checking for differing emoji

### PR DESCRIPTION
**Pull request by Eejit43**
This PR fixes `Activity.equals()` not checking for differing emojis, leading `presenceUpdate` to not fire if only a custom status emoji was updated.

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

File: packages/discord.js/src/structures/Presence.js [hunk 1]
```diff
@@ -254,7 +254,8 @@ class Activity {
         this.type === activity.type &&
         this.url === activity.url &&
         this.state === activity.state &&
-        this.details === activity.details)
+        this.details === activity.details &&
+        this.emoji === activity.emoji)
```
Comment (almeidx) [inline packages/discord.js/src/structures/Presence.js:258]: Something like this is probably a better idea
```suggestion
        this.emoji?.id === activity.emoji?.id)
```

**Review by almeidx — CHANGES_REQUESTED** [review]

File: packages/discord.js/src/structures/Presence.js (hunk 1 above)
Comment (Eejit43) [reply packages/discord.js/src/structures/Presence.js]: Didn't even consider that, thanks! 👍

Comment (RedGuy12) [reply packages/discord.js/src/structures/Presence.js]: don't built in emojis not have an id?

File: packages/discord.js/src/structures/Presence.js [hunk 2]
```diff
@@ -254,7 +254,8 @@ class Activity {
         this.type === activity.type &&
         this.url === activity.url &&
         this.state === activity.state &&
-        this.details === activity.details)
+        this.details === activity.details &&
+        this.emoji?.id === activity.emoji?.id)
```
Comment (GoldenAngel2) [inline packages/discord.js/src/structures/Presence.js:258]: This won't fix the overall problem, if the user changes their custom status from one unicode emoji to another then it won't fire the event either, so both the `emoji?.name` and `emoji?.id` should be checked

**Review by GoldenAngel2 — CHANGES_REQUESTED** [review]

File: packages/discord.js/src/structures/Presence.js (hunk 2 above)
Comment (jaw0r3k) [reply packages/discord.js/src/structures/Presence.js:258]: ^ exactly

Comment (Eejit43) [reply packages/discord.js/src/structures/Presence.js:258]: you're right... wonder why it worked for me in testing then.

however, what if the emoji's name is updated, won't that trigger a false update?

Comment (jaw0r3k) [reply packages/discord.js/src/structures/Presence.js:258]: Your can't prevent that

Comment (Eejit43) [reply packages/discord.js/src/structures/Presence.js:258]: Alright I'll just check for `name` & `id` then 👍

Comment (Eejit43) [reply packages/discord.js/src/structures/Presence.js]: > Alright I'll just check for `name` & `id` then 👍

Better yet, might as well properly check for it: `this.emoji?.id ? this.emoji.id === activity.emoji?.id : this.emoji?name === activity.emoji?.name`

Comment (jaw0r3k) [reply packages/discord.js/src/structures/Presence.js]: idi

Comment (Eejit43) [reply packages/discord.js/src/structures/Presence.js]: > idk

Well I've tested it, it seems to work!

File: packages/discord.js/src/structures/Presence.js [hunk 3]
```diff
@@ -254,7 +254,8 @@ class Activity {
         this.type === activity.type &&
         this.url === activity.url &&
         this.state === activity.state &&
-        this.details === activity.details)
+        this.details === activity.details &&
+        (this.emoji?.id ? this.emoji.id === activity.emoji?.id : this.emoji?.name === activity.emoji?.name))
```
Comment (Jiralite) [inline packages/discord.js/src/structures/Presence.js:258]:
```suggestion
        this.emoji?.id === activity.emoji?.id &&
        this.emoji?.name === activity.emoji?.name)
```

Comment (Eejit43) [reply packages/discord.js/src/structures/Presence.js:258]: Won't this falsely update if the emoji is a guild emoji and its name is updated?

Comment (Jiralite) [reply packages/discord.js/src/structures/Presence.js:258]: How is that a false update?

Comment (Eejit43) [reply packages/discord.js/src/structures/Presence.js:258]: I'm kinda wavering back and forth, but you would know better up to you

Comment (Jiralite) [reply packages/discord.js/src/structures/Presence.js:258]: I would say to change it to this. By the way, modifying the name of a custom emoji will not fire a `PRESENCE_UPDATE` event.

Comment (jaw0r3k) [reply packages/discord.js/src/structures/Presence.js:258]: but updating presence yes
and some old presence can have old name

**Review by kyranet — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Pull request merged**